src/builtins/string.cpp

```diff
@@ -32,10 +32,10 @@
 #include "../wildcard.h"
 #include "../wutil.h"  // IWYU pragma: keep
 
-// How many bytes we read() at once.
-// Bash uses 128 here, so we do too (see READ_CHUNK_SIZE).
-// This should be about the size of a line.
-#define STRING_CHUNK_SIZE 128
+// Empirically determined.
+// This is probably down to some pipe buffer or some such,
+// but too small means we need to call `read(2)` and str2wcstring a lot.
+#define STRING_CHUNK_SIZE 1024
```

Comment:
Makes sense, LGTM.
It's probably important that the buffer doesn't exceed the pagesize (4096 here).

READ_CHUNK_SIZE is still at 128 but that's okay.

> jq '.[]' names.json | string trim -c '"' >/tmp/all

can probably use `jq --raw-output` instead of `string trim`

Reply:
>READ_CHUNK_SIZE is still at 128 but that's okay.

Oh, sure, my idea was that `read` isn't really useful on long inputs, because it only ever reads up to the next newline/NULL.

So you need *one line* that exceeds the 128 bytes to trigger it. That's not *impossible*, but it seems it's not too common.

>can probably use jq --raw-output instead of string trim

To be honest I use jq once every blue moon or so, and I find the syntax hard to remember. I happened to have the `'.[]'` bit stored in my history, so I tried it and noticed it had quotes so I trimmed them.